mesonbuild/interpreter/mesonmain.py

```diff
@@ -54,6 +54,11 @@ def __init__(self, build: 'build.Build', interpreter: 'Interpreter'):
         super().__init__(subproject=interpreter.subproject)
         self.build = build
         self.interpreter = interpreter
+
+        # Override find_program('meson') to return what we were invoked with
+        meson = OverrideProgram('meson', self.interpreter.environment.get_build_command())
```

Comment:
I am not sure if this is the best place for the override, but it seems pretty convenient.

Reply:
It should be contained within the implementation of find_program -- the python3 command is also special cased there, as a fallback.